### PR TITLE
[docs] Tests and guidance for phpunit / Functional tests

### PR DESCRIPTION
**Pull request by cosmicdreams**
## The Issue
Went through the process of getting phpunit + test debugging working for Functional tests and found that some additional documentation is needed to guide people through that process.  

This probably isn't the final documentation that will be committed but I want to get this in front of you as a starting point.

## How This PR Solves The Issue
Additional documentation is needed to support developers who are trying to get phpunit working.  Maybe this makes sense outside of the context of getting phpstorm working. 

## Obsolete
This issue was marked as a duplicate of https://github.com/ddev/ddev/pull/4732

<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/4693"></a>

### CODE REVIEW

File: docs/content/users/install/phpstorm.md [hunk 1]
```diff
@@ -64,6 +73,36 @@ This assumes you’ll need to use PHPUnit and you already have it installed.
...
+### FunctionalJavascript
+```bash
+ddev exec phpunit -c /var/www/html/web/core/phpunit.xml.dist /var/www/html/web/core/modules/node/tests/src/FunctionalJavascript/NodePreviewLinkTest.php
+```
+
+Each of these tests should run properly without any errors.  
+
+## Troublshooting
+### File permission issues when running Functional Tests.
+
```
Comment (cosmicdreams) [inline docs/content/users/install/phpstorm.md:104]: thought I wrote more here.

**Review by cosmicdreams — COMMENTED** [review]
Needs work.

**Comment by cosmicdreams** [comment]
Missing documentation resolved.

**Comment by rfay** [comment]
This one is obsolete now right?

**Comment by cosmicdreams** [comment]
This is obsolete in that the focus for improvement is in https://github.com/ddev/ddev/pull/4732

But I'd love to save the troubleshooting and sample tests to run from this change.  Thoughts on including those changes?

**Comment by rfay** [comment]
I don't agree about keeping Troubleshooting in its current form, but the sample tests are super useful.

**Comment by rfay** [comment]
Why don't you do a commit to remove the obsolete stuff and (current) Troubleshooting. Keep the verification techniques. I'm totally fine with troubleshooting of course, but don't think keeping up with Docker Desktop bugs helps much.

Don't forget to update the PR description.

**Comment by rfay** [comment]
This issues have been sorted out in the PhpStorm extension, and DDEV no longer has a /var/tmp/html in HEAD.

I'd love if it you could take a look at current HEAD DDEV and see what still needs to be done. https://ddev.readthedocs.io/en/latest/developers/building-contributing/#testing-latest-commits-on-head

Closing, happy to continue the conversation here or elsewhere, and thanks for your initiative to get this all sorted out!

**Pull request closed without merging**